cmake/compiler_config.cmake

```diff
@@ -136,5 +136,6 @@ function(mayaHydra_compile_config TARGET)
         PRIVATE
             TBB_SUPPRESS_DEPRECATED_MESSAGES # Remove TBB deprecation warnings
             BOOST_ALL_NO_LIB # Avoid Boost autolinking libraries
+            $<$<STREQUAL:${CMAKE_BUILD_TYPE},Debug>:BOOST_DEBUG_PYTHON> # Auto-select Boost python version
```

Comment:
Why do we need this?  We shouldn't have any Boost Python.

Reply:
(updated since then, see [this comment](https://github.com/Autodesk/maya-hydra/pull/34#discussion_r1451023214))